xcm/src/v3/junction.rs

```diff
@@ -231,6 +231,7 @@ pub enum Junction {
 	/// An instanced, indexed pallet that forms a constituent part of the context.
 	///
 	/// Generally used when the context is a Frame-based chain.
+	// TODO XCMv4 inner should be `Compact<u32>`.
```

Comment:
Can we ever have a runtime that has over 256 pallets? If not, then I'm not sure why it has to be u32 then.

Reply:
"ever" is a long time, and there's no great reason why there should not be more than 256 pallet instances.